govcd/vapp_network_test.go

```diff
@@ -0,0 +1,150 @@
+		&types.FirewallRule{Description: "myFirstRule2", IsEnabled: false, Policy: "drop", DestinationPortRange: "Any",
+			DestinationIP: "Any", SourcePortRange: "Any", SourceIP: "Any", Protocols: &types.FirewallRuleProtocols{Any: true}}}, "drop", true)
+	check.Assert(err, IsNil)
+	check.Assert(result, NotNil)
+
+	// verify
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[0].Description, Equals, "myFirstRule1")
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[0].IsEnabled, Equals, true)
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[0].Policy, Equals, "allow")
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[0].DestinationPortRange, Equals, "Any")
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[0].DestinationIP, Equals, "Any")
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[0].SourcePortRange, Equals, "Any")
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[0].SourceIP, Equals, "Any")
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[0].Protocols.TCP, Equals, true)
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[0].Protocols.TCP, Equals, true)
+
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[1].Description, Equals, "myFirstRule2")
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[1].IsEnabled, Equals, false)
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[1].Policy, Equals, "drop")
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[1].DestinationPortRange, Equals, "Any")
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[1].DestinationIP, Equals, "Any")
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[1].SourcePortRange, Equals, "Any")
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[1].SourceIP, Equals, "Any")
+	check.Assert(result.Configuration.Features.FirewallService.FirewallRule[1].Protocols.Any, Equals, true)
+
+	check.Assert(result.Configuration.Features.FirewallService.DefaultAction, Equals, "drop")
+	check.Assert(result.Configuration.Features.FirewallService.LogDefaultAction, Equals, true)
+
+	//cleanup
+	task, err := vapp.Delete()
+	check.Assert(err, IsNil)
+	err = task.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+	check.Assert(task.Task.Status, Equals, "success")
+}
+
+func (vcd *TestVCD) prepareVappWithVppNetwork(check *C, vappName string) (*VApp, string, *types.NetworkConfigSection, error) {
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	vapp, err := createVappForTest(vcd, vappName)
+	check.Assert(err, IsNil)
+	check.Assert(vapp, NotNil)
+
+	networkName := "Test_UpdateNetworkFirewallRules"
+	description := "Created in test"
+	const gateway = "192.168.0.1"
```

Comment:
Why do we have all these constants if then we don't use any of them to check the results?
If the value is used only once, it's better to use a literal in the declaration of `vappNetworkSettings`.

Reply:
moved to vappNetworkSettings creation.